src/world/partner/goombario.c

```diff
@@ -25,11 +29,110 @@ void world_goombario_init(Npc* partner) {
     partner->collisionRadius = 20;
 }
 
-INCLUDE_ASM(ApiStatus, "world/partner/goombario", func_802BD188_3170A8, Evt* script, s32 isInitialCall);
+ApiStatus func_802BD188_3170A8(Evt* script, s32 isInitialCall) {
+    Npc* npc = script->owner2.npc;
+
+    D_802BDF60 = -1;
+    if (isInitialCall) {
+        partner_init_get_out(npc);
+    }
+
+    if (partner_get_out(npc) != 0) {
+        return ApiStatus_DONE1;
+    } else {
+        return ApiStatus_BLOCK;
+    }
+}
 
-INCLUDE_ASM(ApiStatus, "world/partner/goombario", func_802BD1D0_3170F0, Evt* script, s32 isInitialCall);
+s32 func_802BD1D0_3170F0(Evt* script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+    Entity* temp_s2;
+    Npc* npc;
+    f32 sp10, sp14, tempY;
+
+    npc = script->owner2.npc;
+    if (isInitialCall != 0) {
```

Comment:
no != 0, npc init can go with the declaration right after playerData

Reply:
@Rainchus you resolved this conversation without removing the `!= 0` or moving the declaration up, so I opened it up again

Reply:
My bad, I see you resolved them before pushing your new changes yet!